src/post.js

```diff
@@ -57,9 +70,38 @@ module.exports = function (config, cb) {
         })
       }
 
-      github.gitdata.createReference(tag, function (err) {
+      github.gitdata.createReference(versionTag, function (err) {
         if (err) return cb(err)
 
+        // create reference based on the custom tag
+        var customReference = {
+          owner: ghRepo[0],
+          repo: ghRepo[1],
+          ref: refPart + '/' + customTagName
+        }
+
+        // try to retrive it
+        github.gitdata.getReference(customReference, function (err, ref) {
+          // if it doesn't exist,
+          if (err) {
+            // try to create it
+            github.gitdata.createReference(customTag, function (err) {
+              if (err) return cb(err)
+            })
+          } else if (ref) {
+            // if reference exists, try to delete it
+            github.gitdata.deleteReference(customReference, function (err, msg) {
+              if (err) return cb(err)
+              else if (msg) {
+                // if delete was successful, try to create again
+                github.gitdata.createReference(customTag, function (err) {
+                  if (err) return cb(err)
+                })
+              }
+            })
+          }
+        })
```

Comment:
This part makes sure the reference (tag) doesn't exist when we create it, otherwise, it throws an error. But on a first glance, I believe I forgot to check whether the custom tag was defined in the first place.

Reply:
Isn't the github API would already throw an error when you create a tag that already exist?

Reply:
Exactly, and thus deleting it when that's the case.

Reply:
What I mean is:
- We probably don't want to delete tags. We have no way to check if the original tag is intentional or not. We don't want to take any destructive action without user consent.
- If the objective is to test if the tag exist before creating it, the "create tag" API call is already doing that.

Reply:
I agree with you. Originally I thought the same tag could have multiple versions within, but I believe that's not possible and thus the need of replacing it.

Would `next/v2.0.0` as you mentioned earlier make more sense?